collect_app/src/main/java/org/odk/collect/android/widgets/utilities/RangeWidgetDataRequester.java

```diff
@@ -0,0 +1,186 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.material.slider.Slider;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.RangeQuestion;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.R;
+import org.odk.collect.android.databinding.RangeWidgetHorizontalBinding;
+import org.odk.collect.android.databinding.RangeWidgetVerticalBinding;
+import org.odk.collect.android.fragments.dialogs.NumberPickerDialog;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.ToastUtils;
+import org.odk.collect.android.utilities.WidgetAppearanceUtils;
+import org.odk.collect.android.views.TrackingTouchSlider;
+
+import java.math.BigDecimal;
+
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.NO_TICKS_APPEARANCE;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.VERTICAL_APPEARANCE;
+
+public class RangeWidgetDataRequester {
+
+    private RangeWidgetDataRequester() {
+    }
+
+    public static class RangeWidgetLayoutElements {
+        private final View answerView;
+        private final TrackingTouchSlider slider;
+        private final TextView currentValue;
+        private final TextView minValue;
+        private final TextView maxValue;
+
+        public RangeWidgetLayoutElements(View answerView, TrackingTouchSlider slider,
+                                         TextView currentValue, TextView minValue, TextView maxValue) {
+            this.answerView = answerView;
+            this.slider = slider;
+            this.currentValue = currentValue;
+            this.minValue = minValue;
+            this.maxValue = maxValue;
+        }
+
+        public View getAnswerView() {
+            return answerView;
+        }
+
+        public TrackingTouchSlider getSlider() {
+            return slider;
+        }
+
+        public TextView getCurrentValue() {
+            return currentValue;
+        }
+
+        public TextView getMinValue() {
+            return minValue;
+        }
+
+        public TextView getMaxValue() {
+            return maxValue;
+        }
+    }
+
+    public static RangeWidgetLayoutElements getLayoutElements(Context context, String appearance) {
```

Comment:
All these static helpers feel like they don't fit here? I think they should just move back to a `...Utils` class.